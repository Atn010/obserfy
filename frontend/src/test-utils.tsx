/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
