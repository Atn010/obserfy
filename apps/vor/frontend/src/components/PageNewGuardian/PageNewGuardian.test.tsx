import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./PageNewGuardian.stories"

describe("PageNewGuardian", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
