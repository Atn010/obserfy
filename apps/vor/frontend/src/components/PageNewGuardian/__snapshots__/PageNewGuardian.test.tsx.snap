// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNewGuardian should render correctly 1`] = `
.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-27 {
    font-size: 20px;
  }
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin: 16px;
  height: 24px;
  width: 24px;
  margin-right: 4px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  text-rendering: optimizeLegibility;
  --webkitFontSmoothing: antialiased;
  --mozOsxFontSmoothing: grayscale;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  font-size: 16px;
  margin-bottom: 0;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 14px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 2.5rem;
  line-height: 3rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  text-rendering: optimizeLegibility;
  --webkitFontSmoothing: antialiased;
  --mozOsxFontSmoothing: grayscale;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  margin-left: 16px;
  margin-right: 16px;
  margin-bottom: 16px;
  margin-top: 16px;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  as: label;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 8px;
  color: inherit;
  background-color: transparent;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  background-color: var(--theme-ui-colors-surface,#fff);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  width: 100%;
}

.emotion-5:hover,
.emotion-5:focus {
  border-color: var(--theme-ui-colors-primary,#00e399);
  outline: none;
}

.emotion-5::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-5::-moz-placeholder {
  font-style: italic;
}

.emotion-5:-ms-input-placeholder {
  font-style: italic;
}

.emotion-5::placeholder {
  font-style: italic;
}

.emotion-5:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  color: var(--theme-ui-colors-textDisabled,rgba(0,0,0,0.38));
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  as: label;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 8px;
  color: inherit;
  background-color: transparent;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  background-color: var(--theme-ui-colors-surface,#fff);
  font-size: 16px;
  padding: 12px;
}

.emotion-9:hover,
.emotion-9:focus {
  border-color: var(--theme-ui-colors-primary,#00e399);
  outline: none;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
  width: 24px;
  height: 24px;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  as: label;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  as: label;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 8px;
  color: inherit;
  background-color: transparent;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  background-color: var(--theme-ui-colors-surface,#fff);
  font-size: 16px;
  padding: 16px;
  height: 180px;
  resize: vertical;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  margin-bottom: 16px;
}

.emotion-22:hover,
.emotion-22:focus {
  border-color: var(--theme-ui-colors-primary,#00e399);
  outline: none;
}

.emotion-22::-webkit-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-22::-moz-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-22:-ms-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-22::placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#00e399);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-onPrimary,#000);
  background-color: var(--theme-ui-colors-primary,#00e399);
  border-radius: 8px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-24:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-24:hover,
.emotion-24:focus {
  background-color: var(--theme-ui-colors-primaryDark,#00a06d);
  outline: none;
}

.emotion-24:hover:disabled,
.emotion-24:focus:disabled {
  background-color: var(--theme-ui-colors-primary,#00e399);
}

<div>
  <div
    class="emotion-27"
  >
    <div
      class="emotion-26"
    >
      <a
        href="/dashboard/observe/students/profile/guardians/edit?id=undefined"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-0"
          />
          <p
            class="emotion-1"
          >
            Edit Guardians
          </p>
        </div>
      </a>
      <h5
        class="emotion-3"
      >
        New Guardian
      </h5>
      <div
        class="emotion-25"
      >
        <label
          class="emotion-7"
        >
          <div
            class="emotion-4"
          >
            Guardian Name
          </div>
          <div
            class="emotion-6"
          >
            <input
              class="emotion-5"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-12"
        >
          <div
            class="emotion-4"
          >
            Relationship
          </div>
          <div
            class="emotion-11"
          >
            <select
              aria-label="Relationship"
              class="emotion-9"
            >
              <option
                value="0"
              >
                Other
              </option>
              <option
                value="1"
              >
                Mother
              </option>
              <option
                value="2"
              >
                Father
              </option>
            </select>
            <svg
              class="emotion-10"
              fill="currentcolor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </label>
        <label
          class="emotion-7"
        >
          <div
            class="emotion-4"
          >
            Email
          </div>
          <div
            class="emotion-6"
          >
            <input
              class="emotion-5"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-20"
        >
          <div
            class="emotion-4"
          >
            Phone
          </div>
          <div
            class="emotion-6"
          >
            <input
              class="emotion-5"
              type="phone"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-23"
        >
          <div
            class="emotion-4"
          >
            Note
          </div>
          <textarea
            aria-label="Note"
            class="emotion-22"
          />
        </label>
        <button
          class="emotion-24"
          disabled=""
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
